### PR TITLE
fix(iOS): title and title color handling added for refresh control

### PR DESCRIPTION
**Pull request by shubhamguptadream11**
## Summary:

Solve a part of this issue:  https://github.com/facebook/react-native/issues/46631

## Changelog:

[IOS] [CHANGED] - Passed correct title and titleColor prop to updateTitle function

**What's the Issue:**
When updating the PullToRefreshViewProps in a React Native iOS app, changes to the title and titleColor were not being reflected properly in the RefreshControl. This happened because the function responsible for updating the title (_updateTitle) was not always receiving the correct or updated values for title and titleColor.

**Updated `_updateTitle` function:**

The _updateTitle method was modified to accept both title and titleColor as parameters. This ensures that the latest values are always used when updating the refresh control's attributedTitle.
If the title is empty, the attributedTitle is cleared by setting it to nil. Otherwise, both the title and titleColor (if present) are applied correctly.

## Test Plan:

**Without fix:**
https://github.com/user-attachments/assets/8a83c247-bf78-4080-bdc1-ac5a852481e8

**With Fix:**
https://github.com/user-attachments/assets/52e2495a-4419-41d1-b308-acb64600f9f7

### CODE REVIEW

**Comment by cortinico** [comment]
Is this a duplicate of https://github.com/facebook/react-native/pull/46628 ?

File: packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm [hunk 1]
```diff
@@ -89,7 +89,7 @@ - (void)updateProps:(const Props::Shared &)props oldProps:(const Props::Shared &
   }
 
   if (needsUpdateTitle) {
-    [self _updateTitle];
+    [self _updateTitle:newConcreteProps.title titleColor:newConcreteProps.titleColor];
   }
```
Comment (cipolleschi) [inline packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm:93]: I think it is enough to move the 

```
if (needsUpdateTitle) {
    [self _updateTitle];
  }
```

after calling `super updateProps:props oldProps:oldProps];`

**Review by cipolleschi — CHANGES_REQUESTED** [review]
I believe that this PR adds too many unnecessary changes that will make the code harder to maintain. I left a comment to simplify it dramatically.

File: packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm (hunk 1 above)
Comment (shubhamguptadream11) [reply packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm]: Thank you for your feedback! I understand your point about the unnecessary complexity. I appreciate the suggestion to simplify the logic by moving the if (needsUpdateTitle) block after the super updateProps:props oldProps:oldProps]; call.

I had changed the PR accordingly. Please check again.

**Review by cipolleschi — APPROVED** [review]
Thanks for fixing this!

**Pull request closed without merging**